test/integration/protocol_integration_test.cc

```diff
@@ -467,25 +467,6 @@ TEST_P(ProtocolIntegrationTest, PeriodicAccessLog) {
   EXPECT_EQ("200", response->headers().getStatusValue());
 }
 
-// Regression test for https://github.com/envoyproxy/envoy/issues/9873
-TEST_P(ProtocolIntegrationTest, ResponseWithHostHeader) {
-  initialize();
-  codec_client_ = makeHttpConnection(lookupPort("http"));
-  auto response = codec_client_->makeHeaderOnlyRequest(
-      Http::TestRequestHeaderMapImpl{{":method", "GET"},
-                                     {":path", "/test/long/url"},
-                                     {":scheme", "http"},
-                                     {":authority", "sni.lyft.com"}});
-  waitForNextUpstreamRequest();
-  upstream_request_->encodeHeaders(
-      Http::TestResponseHeaderMapImpl{{":status", "200"}, {"host", "host"}}, true);
-  ASSERT_TRUE(response->waitForEndStream());
-  EXPECT_TRUE(response->complete());
-  EXPECT_EQ("200", response->headers().getStatusValue());
-  EXPECT_EQ("host",
-            response->headers().get(Http::LowerCaseString("host"))[0]->value().getStringView());
-}
```

Comment:
I'm surprised to see this regression test removed. Can you say more about why we're removing it?

Reply:
https://source.chromium.org/chromium/chromium/src/+/main:net/third_party/quiche/src/quiche/http2/http2_constants.cc;l=158?q=http2_constants.cc&ss=chromium here it seems that "host" is not allowed. Maybe I should only disable it for http3.